mmpose/datasets/datasets/top_down/topdown_mpii_dataset.py

```diff
@@ -0,0 +1,227 @@
+
+            # Adjust center/scale slightly to avoid cropping limbs
+            if center[0] != -1:
+                center[1] = center[1] + 15 * scale[1]
+                scale = scale * 1.25
+
+            # MPII uses matlab format, index is 1-based,
+            # we should first convert to 0-based index
+            center = center - 1
+
+            joints_3d = np.zeros((self.ann_info['num_joints'], 3),
+                                 dtype=np.float)
+            joints_3d_visible = np.zeros((self.ann_info['num_joints'], 3),
+                                         dtype=np.float)
+            if not self.test_mode:
+                joints = np.array(a['joints'])
+                joints[:, 0:2] = joints[:, 0:2] - 1
+                joints_vis = np.array(a['joints_vis'])
+                assert len(joints) == self.ann_info['num_joints'], \
+                    f'joint num diff: {len(joints)}' + \
+                    f' vs {self.ann_info["num_joints"]}'
+
+                joints_3d[:, 0:2] = joints[:, 0:2]
+                joints_3d_visible[:, 0] = joints_vis[:]
+                joints_3d_visible[:, 1] = joints_vis[:]
+
+            gt_db.append({
+                'image_file':
+                os.path.join(self.img_prefix, image_name),
+                'center':
+                center,
+                'scale':
+                scale,
+                'rotation':
+                0,
+                'joints_3d':
+                joints_3d,
+                'joints_3d_visible':
+                joints_3d_visible,
+                'dataset':
+                'mpii',
+                'bbox_score':
+                1
+            })
+
+        return gt_db
+
+    def evaluate(self, outputs, res_folder, metrics='PCKh', **kwargs):
+        """Evaluate PCKh for MPII dataset. Adapted from
+        https://github.com/leoxiaobin/deep-high-resolution- net.pytorch
+        Original licence: Copyright (c) Microsoft, under the MIT License.
+
+        Note:
+            num_keypoints: K
+
+        Args:
+            outputs(list(preds, boxes, image_path)):Output results.
+                preds(np.ndarray[1,K,3]): The first two dimensions are
+                    coordinates, score is the third dimension of the array.
+                boxes(np.ndarray[1,6]): [center[0], center[1], scale[0]
+                    , scale[1],area, score]
+                image_path(list[str]): For example, ['0', '0',
+                    '0', '0', '0', '1', '1', '6', '3', '.', 'j', 'p', 'g']
+            res_folder(str): Path of directory to save the results.
+            metrics(str): Metrics to be performed.
+                Defaults: 'PCKh'.
+
+        Returns:
+            PCKh for each joint
+        """
+
+        # only PCKh is supported.
+        assert metrics == 'PCKh'
+
+        _preds = []
+        for i in range(len(outputs)):
+            kpts, _, _ = outputs[i]
+            _preds.append(kpts[0])
+
+        preds = np.stack(_preds)
```

Comment:
actually, this can be merged with above line `preds = np.stack([kpts[0] for kpts, _, _ in outputs])`